app/Controllers/javascriptController.php

```diff
@@ -43,7 +43,11 @@ public function nonceAction() {
 		} else {
 			Minz_Log::notice('Nonce failure due to invalid username!');
 		}
-		$this->view->nonce = '';	//Failure
-		$this->view->salt1 = '';
+		//Failure: Return random data.
+		$this->view->salt1 = sprintf('$2a$%02d$', FreshRSS_user_Controller::BCRYPT_COST);
+		for ($i = 22; $i > 0; $i--) {
+			$this->view->salt1 .= './ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'[rand(0, 63)];
```

Comment:
Well spotted. It is indeed first PHP 5.5 that added support for accessing characters within string literals using []...
http://php.net/manual/en/language.types.string.php